modules/javafx.graphics/src/main/java/javafx/application/Platform.java

```diff
@@ -391,6 +391,18 @@ public static boolean isNestedLoopRunning() {
         return Toolkit.getToolkit().isNestedLoopRunning();
     }
 
+    /**
+     * Indicates whether a nested event loop can be started from the current thread in the current state.
+     * Note that a nested event loop is not allowed outside an event handler.
```

Comment:
You need to be explicit about whether or not this method must be called on the JavaFX thread. The two logical choices are:
1. Add a statement indicating that this method must be called on the JavaFX Application thread. The toolkit method would then call `checkFxUserThread()`, as do the other nested event loop methods.
2. Add a statement indicating that this method may be called on any thread (and that it will return false if not on the JavaFX application thread). The toolkit method would then first call `isFxUserThread()` and return  false if not.

I might lean towards option 2, but don't have a strong preference.

Reply:
Should I also adjust the `Toolkit` javadoc or only the `Platform` one?

Reply:
It might be a good idea to also adjust the Toolkit docs; it doesn't get generated as javadoc, but could be useful for someone looking at the code. If you do, it wouldn't be part of the CSR, since Toolkit isn't API.

Reply:
So someone is asking a question when they call this.
Throwing an Exception could be considered an unfriendly way of answering it.
It only makes sense if you think the programmer will have enough information to be sure that if S/HE does things correctly, it will always be the FxUserThread that is being used .. ie if we are in an event handler .. as is supposed to be the case .. and anything else is a blatant programming error that should be rewarded with an upfront response to that effect.

So
1) programming error -> Exception
2) Can't be sure -> return false.

Reply:
@prrace What is your preference here?
Since `enterNestedEventLoop` and `exitNestedEventLoop` throw an exception when not on the fx user thread, I feel like it makes sense to also add it here as well.

Reply:
Phil's comment got me thinking about this further. All other event loop methods, including `isNestedLoopRunning`, need to be called on the JavaFX thread. So I prefer option 1 as well for consistency.

Reply:
Yes, I agree.